src/main/java/io/strimzi/kafka/bridge/mqtt/core/MqttServerHandler.java

```diff
@@ -24,6 +29,21 @@
  */
 public class MqttServerHandler extends SimpleChannelInboundHandler<MqttMessage> {
     private static final Logger logger = LoggerFactory.getLogger(MqttServerHandler.class);
+    private MqttKafkaMapper mqttKafkaMapper;
+
+    /**
+     * Constructor
+     */
+    public MqttServerHandler() {
+        super(false);
+        try {
+            MappingRulesLoader mappingRulesLoader = MappingRulesLoader.getInstance();
```

Comment:
Not sure I understand the purpose `.getInstance()` method, couldn't we just create a MappingRulesLoader instance by doing:
```suggestion
            MappingRulesLoader mappingRulesLoader = new MappingRulesLoader();
```

Reply:
No, the MappingRulesLoader is responsible for loading the rules from a file. It needs the file path. This path is set only once during the bridge startup(--mapping-rules=path/to/file).

Please take a look at [this line](https://github.com/strimzi/strimzi-mqtt-bridge/pull/8/files#:~:text=MappingRulesLoader.getInstance().setMapperRuleFilePath(mappingRulesFile)%3B)
I really had a harsh time finding the best way to adopt this.

But here are some reasons to adopt singleton in this scenario.
1. I do not want to pass the rules path from the Main class to other classes. This path is provided in the args --mapping-rules=path/to/file, if I give the responsibility to load the mapping rules file to MqttConfig, I will have to pass the loaded rules as an argument to the MqttServer class, then to the MqttServerInitializer class and finally to the MqttServerHandler class which has the MqttKafkaMapper instance. So it would be hard to maintain and track the lifecycle of the mapping rules.

2. Single responsibility principle, I want to make the project clear and easy maintainable. For the entire lifecycle of the bridge, we just need one instance of the MappingRulesLoader, set the path from the very beginning of the bridge startup and use the loadRules() anywhere in the project.

3. Again single responsibility, right now if news rules are added while the bridge is already running, we won't get the fresh rules, to get it, we need to restart the server. Even if we are not considering this case now, I thought it would be better to implement something that will make it easy to change the scenario. For example, in future we may want to keep track of the mapping rules file, check for new entries and update the rules inside the bridge, we would just add all the logic in the MappingRulesLoader.

So to answer the question, if we do
MappingRulesLoader = new MappingRulesLoader();
we will need to set the path which would come from the Main class.

Reply:
While I am not totally against the idea of having a singleton pattern, I would avoid having a setter to provide the path because it would mean you could set it whenever through the code, while I would leave the singleton "configuration" immutable. I would have an `init` method (where you can provide the file path) which would raise and exception if called one more time.

Reply:
I have done that!

Reply:
I am waiting for @kyguy to say something regarding this conversation.

Reply:
That makes sense @antonio-pedro99 good forethought, thanks for explaining!